llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -1607,11 +1607,20 @@ ChangeStatus AAPointerInfoFloating::updateImpl(Attributor &A) {
     //
     // The RHS is a reference that may be invalidated by an insertion caused by
     // the LHS. So we ensure that the side-effect of the LHS happens first.
+
+    if (!OffsetInfoMap.contains(Usr)) {
```

Comment:
Why this check? Can't we merge into an empty value?

Reply:
I tried an assertion there like you had suggested, but that causes that assertion to fail at runtime for multiple tests.